code/modules/chemistry/Reagents-ExplosiveFire.dm

```diff
@@ -720,10 +720,17 @@ datum
 				return 1
 
 			on_mob_life(var/mob/M, var/mult = 1)
-
-				var/mob/living/L = M
-				if(istype(L) && L.getStatusDuration("burning"))
-					L.changeStatus("burning", 2 SECONDS * mult)
+				if (!M)
+					M = holder.my_atom
+				if(istype(M, /mob/living/) && M.getStatusDuration("burning"))
+					M.changeStatus("burning", 2 SECONDS * mult)
+				if(holder.has_reagent("welding fuel"))
+					holder.remove_reagent("welding fuel", 1 * mult)
```

Comment:
Add a depletion rate instead of doing this weird thing.

Reply:
Didn't know this existed and reused a thing I found elsewhere.
Which means we have janky code elsewhere not using depletion rate...